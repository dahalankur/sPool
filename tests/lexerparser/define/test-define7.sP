# list of lambdas
list<(int, bool -> int)> listOfLambdas = [lambda int (int i, bool b): return i + 1;, lambda int (int i, bool c): return i + 2 + c;]