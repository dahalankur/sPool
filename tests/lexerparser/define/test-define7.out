PROGRAM {[ 
    DEFINE(false, LISTTY(ARROWTY(INTTY * BOOLTY -> INTTY)), listOfLambdas, LIST(LAMBDA(INTTY, FORMALS((INTTY, i), (BOOLTY, b)), [ 
        RETURN(BINOP(VAR(i), PLUS, INT(1)))]), LAMBDA(INTTY, FORMALS((INTTY, i), (BOOLTY, c)), [ 
        RETURN(BINOP(BINOP(VAR(i), PLUS, INT(2)), PLUS, VAR(c)))])))]}
